#14 Announcements Cell Redesign

konnork opened the pull request:
### What

Redesign for announcements to  show sponsorship label if sponsored and general redesign

![simulator screen shot sep 13 2016 4 09 38 pm](https://cloud.githubusercontent.com/assets/12649134/18489674/7f47feb6-79cc-11e6-9a38-8ae2e6b5313f.png)

Addressed #14 
@russellladd

konnork committed:
Redesigned Announcements TableViewCell

konnork committed:
Added hide on sponsored

russellladd commented:
Looks amazing. I'll look at it more closely and merge today.

russellladd commented on MHacks/AnnouncementCell.swift:
```diff
@@ -16,6 +16,8 @@ class AnnouncementCell: UITableViewCell {
     @IBOutlet weak var messageLabel: UILabel!
 	@IBOutlet weak var colorView: UIView!
+	@IBOutlet weak var sponsoredLabel: UIView!
+	@IBOutlet weak var contentStackView: UIStackView!
```
Please do not use weak for any of these vars and remove it from the ones that already exist. There's no need for these views to be weakly held, and it's a bug if they actually get released. The Sponsored label getting released across cell reuse caused a crash for me.

konnork replied:
Fixed

russellladd commented on MHacks/AnnouncementsViewController.swift:
```diff
@@ -97,16 +97,21 @@ class AnnouncementsViewController: UITableViewController {
+		
+		if !announcement.isSponsored {
+			cell.contentStackView.removeArrangedSubview(cell.sponsoredLabel)
+			cell.sponsoredLabel.removeFromSuperview()
```
This is fine but you could more simply accomplish the same affect by setting the hidden property. UIStackView will correctly not allocate space for hidden views.

konnork replied:
Just like the below comment, thats what I did first but it didn't work :/ I just tried it again and there is noticeably more space being rendered with
`cell.sponsoredLabel.isHidden = !announcement.isSponsored` rather than lines 110 - 113...

I don't know if that means that `UIStackView` only _updates_ the layout when the view is hidden rather than if the view is hidden before the contents of the stack view are even rendered? but that doesn't really make sense why so maybe I'm just doing something dumb

russellladd commented on MHacks/RoundedView.swift:
```diff
@@ -0,0 +1,18 @@
+import UIKit
+
+@IBDesignable
+class RoundedView: UIView {
```
Please don't use IBDesignable. I know it's cool but it makes storyboards unbearably slow on older computers. For this case you can really just add a line of code in the AnnouncementCell initializer that sets the corner radius. Also, the corner radius needs to be set to 1, not 2, since the width of the view is 2.

russellladd replied:
So you could delete this entire class.

manavgabhawala replied:
I'm pretty sure you can set this in the storyboard too, using User Defined Runtime Attributes

russellladd replied:
Yeah you can. But that's built on KVC (sketchy) and is a pretty obscure feature. I'd prefer it written in code for clarity and discoverabily.

konnork replied:
@russellladd That's what I did first but I tried it on simulator and on device and I can't just can't see any rounding unless I put 2 :/

konnork replied:
(I fixed the rest of this conversation)

russellladd reviewed (changes requested):
Please make these changes and request again - thank you!

konnork committed:
Change weak references and roundedview

manavgabhawala committed:
Made the Add to Apple Wallet button actually work

russellladd committed:
Improved ticket layout

russellladd committed:
Merge remote-tracking branch 'origin/master'

manavgabhawala committed:
Updated code to work with new APNS code on backend

manavgabhawala committed:
Implement updateAnnouncement and deleteAnnouncement and remove all the deprecated methods.

manavgabhawala committed:
Added scan events with complete implementation, serialization, and updating from the db

manavgabhawala committed:
Updates to user serializer
Actually implemented privileges
Added the ability to update the profile for a user so that their permissions can be updated

russellladd committed:
Merge remote-tracking branch 'origin/announcements' into announcements

# Conflicts:
#	MHacks/Base.lproj/Main.storyboard
